### DISCUSSION TITLE
Special characters escaping (or otherwise rendering incorrectly)

### DISCUSSION THREAD

**Discussion by JodiWarren** [dandavison/delta#1096] [Q&A]
When diffing 

`$ delta test1.txt test2.txt`

```diff
- hello world
+ hello jodi
```

I get the following output:
```
^[[38;5;88m  1 ⋮^[[38;5;28m    │^[[48;2;63;0;1mhello ^[[48;2;144;16;17mworld^[[48;2;63;0;1m^[[0K
^[[38;5;88m    ⋮^[[38;5;28m  1 │^[[48;2;0;40;0;38;2;248;248;242mhello ^[[48;2;0;96;0mjodi^[[48;2;0;40;0m^[[0K
```

I'm not entirely sure what could be causing this. I've just installed Delta via homebrew on OSX and have set it up with the suggested simple config. I've installed the latest version of `less` as suggested. I've experienced this in both Terminal and iTerm2.

For contrast, `$ bat test1.txt` outputs:

```
───────┬────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: test1.txt
───────┼────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ hello world
───────┴────────────────────────────────────────────────────────────────────────────────────────────────────────
```

**Comment by bew** [comment]
Can you try `$ LESS="-r" delta test1.txt test2.txt` ?
Also, which version of `less` do you have?

**Reply by JodiWarren** [reply]
Hi @bew , thanks so much for taking a look.

`LESS="-r" delta test1.txt test2.txt` gets me the same results as my initial post, unfortunately.

I'm using `less` 590 (installed via homebrew), and I have double checked that the system is using that version.

**Reply by dandavison** [reply]
@JodiWarren if you don't use a pager, does everything display correctly? E.g. `delta --paging=never`

If so, then what about piping that output into less explicitly: e.g. `git log -p | delta | /path/to/correct/version/of/less -R`?

Incidentally, if you ever want to double check the less that git is picking up, you can do strange tricks like this (here telling git to run `which less` instead of the pager) 
```
git -c core.pager='which less' show
```

**Reply by JodiWarren** [reply]
@dandavison 

`delta --paging=never` works perfectly:

```
$ delta test1.txt test2.txt --paging=never

Δ test1.txt ⟶   test2.txt
────────────────────────────────────────────────────────────────────────────────

─────┐
• 1: │
─────┘
  1 ⋮    │hello world
    ⋮  1 │hello jodi
```

`git log -p | delta | less -R` also works perfectly, as does excluding the `-R` flag. `git log -p | delta` has the same issue as my original post unfortunately. So it looks like there's something up with the way that delta is interfacing with my local version of `less`
